biogtr/datasets/microscopy_dataset.py

```diff
@@ -76,43 +82,25 @@ def __init__(
             for video in self.videos
         ]
 
-        if self.chunk:
-            self.chunks = [
-                [i * self.clip_length for i in range(len(video) // self.clip_length)]
-                for video in self.frame_idx
-            ]
-
-            self.chunked_frame_idx, self.label_idx = [], []
-            for i, (split, frame_idx) in enumerate(zip(self.chunks, self.frame_idx)):
-                frame_idx_split = torch.split(frame_idx, self.clip_length)
-                self.chunked_frame_idx.extend(frame_idx_split)
-                self.label_idx.extend(len(frame_idx_split) * [i])
-        else:
-            self.chunked_frame_idx = self.frame_idx
-            self.label_idx = [i for i in range(len(self.videos))]
-
-    def __len__(self):
-        """Get length of dataset.
+        # Method in BaseDataset. Creates label_idx and chunked_frame_idx to be
+        # used in call to get_instances()
+        self.create_chunks()
 
-        Returns:
-            the length of the dataset
-        """
-        return len(self.chunked_frame_idx)
-
-    def no_batching_fn(self, batch):
-        """Collate function used to overwrite dataloader batching function.
+    def get_indices(self, idx):
+        """Retrieves label and frame indices given batch index.
 
         Args:
-            batch: the chunk of frames to be returned
-
-        Returns:
-            the batch
+            idx: the index of the batch.
         """
-        return batch
+        return self.label_idx[idx], self.chunked_frame_idx[idx]
 
-    def __getitem__(self, idx):
+    def get_instances(self, label_idx: List[int], frame_idx: List[int]) -> list[dict]:
```

Comment:
```suggestion
    def get_instances(self, label_idx: list[int], frame_idx: list[int]) -> list[dict]:
```

Consistent typing notation -- just use `list` since we're in py>=3.9